### DISCUSSION TITLE
How to configure the `highlight` subcommand to use the parsers and queries installed via nvim-treesitter?

### DISCUSSION THREAD

**Discussion by ymn64** [tree-sitter/tree-sitter#2717] [Q&A]
I am trying to get the `tree-sitter highlight` command to produce the same syntax highlighting I get in Neovim, I tried setting `parser-directories` in tree-sitter config file to `/path/to/nvim-treesitter` and `/path/to/nvim-treesitter/parser` but it didn't work as this doesn't follow the expected format mentioned here https://tree-sitter.github.io/tree-sitter/syntax-highlighting#paths.
Any clues on how to achieve this?

**Comment by amaanq** [comment]
parser-directories is for grammars

So just move a parser from your nvim installation into the cache dir, on Linux this is ~/.cache/tree-sitter

but, if you don't have the grammar then tree-sitter might not know what language should be used on a file

**Reply by ymn64** [reply]
You mean copying the `.so` files from `/path/to/nvim-treesitter/parser` directory to `~/Library/Caches/tree-sitter/lib` (macOS equivalent of `~/.cache/tree-sitter/lib` on Linux)?
I've tried that but it didn't work, I'm getting the `No language found for path...` message.

**Reply by amaanq** [reply]
> but, if you don't have the grammar then tree-sitter might not know what language should be used on a file

grammar cloned locally*, but yeah

see https://tree-sitter.github.io/tree-sitter/3-syntax-highlighting.html#language-configuration
